test/unittests/commands/excel2json/test_lists.py

```diff
@@ -0,0 +1,417 @@
+        assert isinstance(res, ListRoot)
+        assert res.id_ == "list_id"
+        assert res.labels == {"en": "Listname_en", "de": "Listname_de"}
+        assert len(res.nodes) == 3
+        assert res.nodes[0].id_ == "1"
+        assert res.nodes[1].id_ == "2"
+        assert res.nodes[2].id_ == "3"
+        assert res.nodes[2].sub_nodes[0].id_ == "3.1"
+        assert res.nodes[2].sub_nodes[1].id_ == "3.2"
+        assert res.nodes[2].sub_nodes[1].sub_nodes[0].id_ == "3.2.1"
+        assert res.nodes[2].sub_nodes[1].sub_nodes[1].id_ == "3.2.2"
+
+
+class TestFillIdColumn:
+    def test_to_fill(self) -> None:
+        test_df = pd.DataFrame(
+            {
+                "ID (optional)": [pd.NA, "1", pd.NA, "3", pd.NA, pd.NA, pd.NA, pd.NA],
+                "en_list": ["list_en", "list_en", "list_en", "list_en", "list_en", "list_en", "list_en", "list_en"],
+                "en_1": [pd.NA, "nd_en_1", "nd_en_2", "nd_en_3", "nd_en_3", "nd_en_3", "nd_en_3", "nd_en_3"],
+                "en_2": [pd.NA, pd.NA, pd.NA, pd.NA, "nd_en_3.1", "nd_en_3.2", "nd_en_3.2", "nd_en_3.2"],
+                "en_3": [pd.NA, pd.NA, pd.NA, pd.NA, pd.NA, pd.NA, "nd_en_3.2.1", "nd_en_3.2.2"],
+            }
+        )
+        expected = [
+            "list_en",
+            pd.NA,
+            "nd_en_2",
+            pd.NA,
+            "nd_en_3.1",
+            "nd_en_3.2",
+            "nd_en_3.2.1",
+            "nd_en_3.2.2",
+        ]
+        res = _fill_auto_id_column(test_df, "en")
+        assert res["auto_id"].to_list() == expected
+
+    def test_nothing_to_fill(self) -> None:
+        test_df = pd.DataFrame(
+            {
+                "ID (optional)": ["list_en", "1", "2", "3", "3.1", "3.2", "3.2.1", "3.2.2"],
+            }
+        )
+        expected = pd.DataFrame(
+            {
+                "ID (optional)": ["list_en", "1", "2", "3", "3.1", "3.2", "3.2.1", "3.2.2"],
+                "auto_id": [pd.NA, pd.NA, pd.NA, pd.NA, pd.NA, pd.NA, pd.NA, pd.NA],
+            }
+        )
+        res = _fill_auto_id_column(test_df, "en")
+        assert_frame_equal(res, expected)
+
+
+def test_handle_ids() -> None:
+    test_df = pd.DataFrame(
+        {
+            "ID (optional)": [pd.NA, pd.NA, "2", "3", pd.NA, "5", "6", pd.NA],
+            "en_list": ["list_en", "list_en", "list_en", "list_en", "list_en", "list_en", "list_en", "list_en"],
+            "en_1": ["list_en", "nd_en_1", "nd_en_2", "nd_en_3", "nd_en_4", "nd_en_5", "nd_en_6", "nd_en_7"],
+        }
+    )
+    res = _fill_id_and_parent_id_columns(test_df, "en")
+    assert res["id"].tolist() == ["list_en", "nd_en_1", "2", "3", "nd_en_4", "5", "6", "nd_en_7"]
+
+
+def test_fill_parent_id() -> None:
+    test_df = pd.DataFrame(
+        {
+            "id": [
+                "list_en",
+                "1",
+                "2",
+                "3",
+                "3.1",
+                "3.2",
+                "3.2.1",
+                "3.2.2",
```

Comment:
if you remove the trailing comma and then run auto-format, this will fit on 1 line